### PR TITLE
Add a config flag to enable the rust crypto-sdk

### PR DESCRIPTION
**Pull request by richvdh**
This PR adds an option to `config.json` which will make the js-sdk use the rust crypto sdk, instead of the libolm implementation.

To use it, you need to add something like this to `config.json`:

```
    "features": {
        "feature_rust_crypto": true
    },
```

We don't (yet) have any way to migrate a device between implementations, so the setting that was in use when you log in is persisted to the device; it is *visible* via the labs section but cannot currently be changed:

image

This is part of https://github.com/vector-im/element-web/issues/21972, and enables the functionality added to the js-sdk in https://github.com/matrix-org/matrix-js-sdk/pull/2969.

---
Here's what your changelog entry will look like:

## ✨ Features
 * Add a config flag to enable the rust crypto-sdk ([\#9759](https://github.com/matrix-org/matrix-react-sdk/pull/9759)).

### CODE REVIEW

**Comment by richvdh** [comment]
I'm hoping for an exemption on the strict type checks. The failures are not on lines I have changed.

**Comment by florianduros** [comment]
@richvdh If your merge develop in your branch, the failures will disappear.

**Comment by t3chguy** [comment]
@richvdh the failing CI is non-required. Please see pinned message in Element Web Internal from @andybalaam

**Comment by richvdh** [comment]
> @richvdh If your merge develop in your branch, the failures will disappear.

Empirically not!

**Review by florianduros — APPROVED** [review]

**Review by turt2live — COMMENTED** [review]
dear github mobile, if I hit "send" 4 hours ago, that doesn't mean hold the comment as a pending review until I happen to notice.

Apologies for the late review - it's only a minor point.

File: src/IConfigOptions.ts [hunk 1]
```diff
@@ -191,6 +191,8 @@ export interface IConfigOptions {
         description: string;
         show_once?: boolean;
     };
+
+    use_rust_crypto_sdk?: boolean;
```
[the remark this answers is not in the document]
Comment (richvdh) [reply src/IConfigOptions.ts:195]: no it's not. bother.

[the remark this answers is not in the document]
Comment (richvdh) [reply src/IConfigOptions.ts:195]: fixed in #9792

**Pull request merged**